pycbc/workflow/dq.py

```diff
@@ -22,183 +22,122 @@
-        node.add_input_opt('--rate-file', binned_rate_file)
-        node.new_output_file_opt(workflow.analysis_time, '.hdf',
-                                 '--output-file')
+        node.add_input_opt('--bank-file', template_bank_file)
+        node.new_output_file_opt(
+            workflow.analysis_time, '.hdf', '--output-file')
         return node
 
+
 class PyCBCBinTriggerRatesDQExecutable(Executable):
     current_retention_level = Executable.MERGED_TRIGGERS
-    def create_node(self, workflow, ifo, dq_files, trig_file, bank_file):
+
+    def create_node(self, workflow, flag_file, flag_name,
+                    analysis_segment_file, analysis_segment_name,
+                    trig_file, template_bins_file):
         node = Node(self)
-        node.add_opt('--ifo', ifo)
-        node.add_input_opt('--bank-file', bank_file)
+        node.add_input_opt('--template-bins-file', template_bins_file)
         node.add_input_opt('--trig-file', trig_file)
-        node.add_input_list_opt('--dq-file', dq_files)
-        node.new_output_file_opt(workflow.analysis_time,'.hdf',
+        node.add_input_opt('--flag-file', flag_file)
+        node.add_opt('--flag-name', flag_name)
+        node.add_input_opt('--analysis-segment-file', analysis_segment_file)
+        node.add_opt('--analysis-segment-name', analysis_segment_name)
+        node.new_output_file_opt(workflow.analysis_time, '.hdf',
                                  '--output-file')
         return node
 
-class PyCBCCalculateDQFlagExecutable(Executable):
-    # current_retention_level = Executable.ALL_TRIGGERS
-    current_retention_level = Executable.MERGED_TRIGGERS
-
-    def create_node(self, workflow, segment, dq_file, flag):
-        node = Node(self)
-        # Executable objects are initialized with ifo information
-        start = int(segment[0])
-        end = int(segment[1])
-        node.add_opt('--ifo', self.ifo_string)
-        node.add_opt('--flag', flag)
-        node.add_opt('--gps-start-time', start)
-        node.add_opt('--gps-end-time', end)
-        node.add_input_opt('--dq-segments', dq_file)
-        node.new_output_file_opt(segment, '.hdf',
-                                 '--output-file')
-        return node
 
-def setup_dq_reranking(workflow, dq_label, insps, bank,
-                        segs, analyzable_file, dq_file,
-                        output_dir=None, tags=None):
+def setup_dq_reranking(workflow, insps, bank,
+                       analyzable_seg_file,
+                       analyzable_name,
+                       dq_seg_file,
+                       output_dir=None, tags=None):
+    logging.info("Setting up dq reranking")
     make_analysis_dir(output_dir)
-    output = FileList()
-    if tags:
+    output_files = FileList()
+    output_labels = []
+    if not tags:
+        tags = []
+
+    dq_labels = workflow.cp.get_subsections('workflow-data_quality')
+    dq_labels = numpy.array(dq_labels)
```

Comment:
I can't help thinking that the block of code from here up to line 96 or so would be easier to read and manipulate by using a dict keyed on dq_labels, and by looping over keys, rather than doing several list comprehensions over the same array (`... for dq_label in dq_labels`).